helix-core/src/path.rs

```diff
@@ -40,6 +40,24 @@ pub fn expand_tilde(path: &Path) -> PathBuf {
 /// needs to improve on.
 /// Copied from cargo: <https://github.com/rust-lang/cargo/blob/070e459c2d8b79c5b2ac5218064e7603329c92ae/crates/cargo-util/src/paths.rs#L81>
 pub fn get_normalized_path(path: &Path) -> PathBuf {
+    // normalization strategy is to canonicalize first ancestor path that exists (i.e., canonicalize as much as possible),
+    // then run handrolled normalization on the non-existent remainder
+    let (base, path) = path
+        .ancestors()
+        .find(|path| path.exists())
+        .and_then(|path| Some((path, dunce::canonicalize(path).ok()?)))
+        .map(|(base, canonicalized)| {
+            (
+                canonicalized,
+                path.strip_prefix(base).unwrap(/* base is an ancestor of path */).into(),
+            )
+        })
```

Comment:
```suggestion
        .find_map(|base| { 
            let canonicalized_base = dunce::canonicalize(base).ok()?;
            Some((canonicalized_base,  path.strip_prefix(base)))
        })
```
just because a path exists doesn't mean cannonicalization necessary successed (Sadly). Using `find_map` ensures we only stoip once canonicalization succedes. It's also a bit cleaner.